sage-ios/SAGE/SAGE/SemesterOperations.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  SemesterOperations.swift
+//  SAGE
+//
+//  Created by Andrew Millman on 1/25/16.
+//  Copyright © 2016 Cal Blueprint. All rights reserved.
+//
+
+import Foundation
+import SwiftKeychainWrapper
+
+class SemesterOperations {
+    
+    static func startSemester(semester: Semester, completion: (Semester) -> Void, failure: (String) -> Void) {
+        let params = [
+            SemesterConstants.kStartDate: semester.dateStringFromStartDate(),
+            SemesterConstants.kTerm: semester.term.rawValue
+        ]
+        
+        BaseOperation.manager().POST(StringConstants.kEndpointStartSemester, parameters: params, success: { (operation, data) -> Void in
+            let semesterDict = (data as! [String: AnyObject])["semester"] as! [String: AnyObject]
+            let createdSemester = Semester(propertyDictionary: semesterDict)
+            completion(createdSemester)
+            }) { (operation, error) -> Void in
+                failure("Could not start semester.")
+        }
+    }
+    
+    static func endSemester(completion: () -> Void, failure: (String) -> Void) {
+        // TODO: Get semester from keychain
+        let semester = Semester()
+        BaseOperation.manager().POST(StringConstants.kEndpointEndSemester(semester.id), parameters: nil, success: { (operation, data) -> Void in
+            completion()
+            }) { (operation, error) -> Void in
+                failure("Could not end semester.")
+        }
+    }
+    
+    static func joinSemester(completion: (() -> Void)?, failure: (String) -> Void) {
+        BaseOperation.manager().POST(StringConstants.kEndpointJoinSemester, parameters: nil, success: { (operation, data) -> Void in
+            let userJSON = data["session"]!!["user"] as! [String: AnyObject]
+            let semesterSummaryJSON = userJSON["user_semester"]
+            var semesterSummary: SemesterSummary? = nil
+            if !(semesterSummaryJSON is NSNull) {
+                semesterSummary = SemesterSummary(propertyDictionary: semesterSummaryJSON as! [String: AnyObject])
+            }
+            KeychainWrapper.setObject(semesterSummary!, forKey: KeychainConstants.kSemesterSummary)
+            completion?()
+            }) { (operation, error) -> Void in
+                failure("Could not join semester.")
```

Comment:
we can actually get the custom messages that charles passes back from BaseOperation.getErrorMessage(error), so you can just do failure(BaseOperation.getErrorMessage(error)). can you change taht for the other operations in this file too, and in the getState function?